plugins/Microsoft.DocAsCode.Build.UniversalReference/ModelConverter/ModelConverter.cs

```diff
@@ -0,0 +1,278 @@
+                .Select(pair =>
+                {
+                    return new ApiLanguageValuePair<List<ApiNames>>
+                    {
+                        Language = pair.Language,
+                        Value = pair.Value.Select(item => ResolveApiNames(item, supportedLanguages, references)).ToList()
+                    };
+                }).ToList();
+        }
+
+        public static List<ApiLanguageValuePair<T>> ToApiListInDevLangs<T>(T defaultValue, SortedList<string, T> values, string[] supportedLanguages)
+        {
+            if (defaultValue == null || supportedLanguages == null || supportedLanguages.Length == 0) return null;
+
+            var result = new List<ApiLanguageValuePair<T>>();
+            foreach (var language in supportedLanguages)
+            {
+                result.Add(new ApiLanguageValuePair<T>
+                {
+                    Language = language,
+                    Value = values.ContainsKey(language) ? values[language] : defaultValue,
+                });
+            }
+
+            return result;
+        }
+
+        public static ApiNames ResolveApiNames(string uid, string[] supportedLanguages, IReadOnlyDictionary<string, ApiNames> references)
+        {
+            ApiNames result;
+            if (references.TryGetValue(uid, out result))
+            {
+                return result;
+            }
+            return new ApiNames
+            {
+                Uid = uid,
+                Spec = supportedLanguages.Select(s => new ApiLanguageValuePair<string>
+                {
+                    Language = s,
+                    Value = GetXref(uid)
+                }).ToList()
+            };
+        }
+
+        private static ApiInheritanceTreeBuildOutput ResolveInheritanceTree(InheritanceTree tree, string[] supportedLanguages, IReadOnlyDictionary<string, ApiNames> references)
+        {
+            return new ApiInheritanceTreeBuildOutput
+            {
+                Type = ResolveApiNames(tree.Type, supportedLanguages, references),
+                Inheritance = tree.Inheritance.Select(item => ResolveInheritanceTree(item, supportedLanguages, references)).ToList(),
+                Metadata = tree.Metadata
+            };
+        }
+
+        private static ApiBuildOutput ResolveApiBuildOutput(string uid, string[] supportedLanguages, IReadOnlyDictionary<string, ApiBuildOutput> children)
+        {
+            ApiBuildOutput result;
+            if (!children.TryGetValue(uid, out result))
+            {
+                var message = $"Can't find {uid} in items or references";
+                Logger.LogError(message);
+                throw new DocumentException(message);
+            }
+            return result;
+        }
+
+        private static string GetXref(string uid, string text = null, string alt = null)
+        {
+            var result = $"<xref uid=\"{HttpUtility.HtmlEncode(uid)}\"";
```

Comment:
should be uid. remove the one in ApiNamesSpecResolver and use this one.

---
In reply to: [114263999](https://github.com/dotnet/docfx/pull/1618#discussion_r114263999) [](ancestors = 114263999)